raiden/transfer/channel.py

```diff
@@ -488,7 +490,9 @@ def is_valid_lock_expired(
     # sure that we check for "unclaimed" locks in our check.
     lock = channel_state.partner_state.secrethashes_to_lockedlocks.get(secrethash)
     if not lock:
-        lock = channel_state.partner_state.secrethashes_to_unlockedlocks.get(secrethash)
+        partial_lock = channel_state.partner_state.secrethashes_to_unlockedlocks.get(secrethash)
```

Comment:
This was the most important problem I found in channel.py thanks to `mypy`. Essentially this worked only because `UnlockPartialProofState` happens to contains the same attributes as `HashTimeLockState` that it contains so at this point in time stuff work. But if that ever changes this would hide a nasty bug.

Note: There is also a test that goes into this codepath. It's [here](https://github.com/raiden-network/raiden/blob/5b4375d24b11f8a2f1032af66c26e5b53684640e/raiden/tests/unit/test_channelstate.py#L1703)